frontend/packages/employee-frontend/src/components/unit/unit-details/UnitEditor.tsx

```diff
@@ -896,6 +912,28 @@ export default function UnitEditor(props: Props): JSX.Element {
           form.costCenter
         )}
       </FormPart>
+      <FormPart>
+        <div>{showRequired(i18n.unitEditor.label.financeDecisionManager)}</div>
+        {props.editable ? (
+          <Select
+            options={employeeOptions}
+            placeholder={i18n.unitEditor.placeholder.financeDecisionManager}
+            onChange={(value) =>
+              value && 'value' in value
+                ? updateForm({ financeDecisionManagerId: value.value })
+                : undefined
+            }
+          />
+        ) : (
+          props.employees
+            .map((employee) => ({
+              id: employee.id,
+              name: [employee.firstName, employee.lastName].join(' ')
+            }))
+            .find((employee) => employee.id === form.financeDecisionManagerId)
```

Comment:
It's usually preferable to first `find` and then `map` the value to whatever format you need.